### PR TITLE
proposal for component curation principles

### PR DESCRIPTION
**Pull request by danmoseley**
Feedback very welcome here.

### CODE REVIEW

File: src/Components/README.md [hunk 1]
```diff
@@ -2,11 +2,29 @@
...
+
+Note: only components that are built from dotnet/aspire will be able to use the Aspire package name prefix. There is no technical barrier to using components built elsewhere, without the Aspire prefix, but currently our idea is that all broadly available Aspire components will live here in dotnet/aspire and have the Aspire package prefix. We welcome feedback on this and all the other principles listed here, though.
+
+In summary we encourage and are excited to accept contributions of components, but it's probably a good idea to first open an issue to discuss any new potential component before offering a PR, to make sure we're all in agreement that it's a good fit with these principles.
+
+## Versioning and Releases
+
+Each component is in its own NuGet package, and can version independently, including declaring itself in a preview state using the standard SemVer and NuGet mechanisms. We expect to typically push updates to all components at the same time we update the core Aspire packages, but we have flexibility here.
```
Comment (DamianEdwards) [inline src/Components/README.md:21]: This should mention that this is restricted by at least the major version, and sometimes the minor depending on timing, e.g. right now we're shipping Aspire 8.0 so we expect all components being added will be versioned with the 8.0.0 prefix.

File: src/Components/README.md [hunk 2]
```diff
@@ -2,11 +2,29 @@
...
-- When component is built around `ABC` client library, it should contain the client library name in its name. Example: `Aspire.ABC`.
-- When given client library is just one of many libraries that allows to consume given service, the names that refer to component need to be specific, not generic. Example: Npgsql is not the only db driver for PostgreSQL database, so the extension method should be called `AddNpgsql` rather than `AddPostgreSQL`.
+- Each component's name must have the prefix `Aspire.`.
+- When component is built around `ABC` client library, it should contain the client library name in its name. Example: `Aspire.ABC`. Where the technology has a particular casing we have preferred that: for example `Aspire.RabbitMQ` rather than `Aspire.RabbitMq`.
+- When the client library is just one of many libraries that allows to consume given service, the names that refer to component need to be specific, not generic. Example: Npgsql is not the only db driver for PostgreSQL database, so the extension method should be called `AddNpgsql` rather than `AddPostgreSQL`. The goal here is to help app authors make an informed decision about which to choose.
```
Comment (DamianEdwards) [inline src/Components/README.md:27]: Can we add something here about a package icon? e.g. that component packages should have a unique icon that reflects the underlying client library or system being integrated with (assuming usage terms allow it).

**Comment by danmoseley** [comment]
updated

**Review by DamianEdwards — APPROVED** [review]

File: src/Components/README.md [hunk 3]
```diff
@@ -2,11 +2,33 @@
...
+
+1. We expect to welcome any components that would have value to Aspire users and align with what Aspire is intended to do, subject to the below.
+2. We don't expect to choose preferred techs. For example, if there are two commonly used providers for database XYZ, we are comfortable with having one component for each. We would like component naming and granularity to be clear enough that customers can make informed decisions. Aspire is agnostic to your choice of cloud provider, too.
+3. We will require that the tech represented by the component is being actively supported. In most cases we expect that it is widely used, although we expect that part will be a judgement call.
+4. Components contributed to dotnet/aspire must meet the same quality and completeness bar of other contributions. ie., we won't have a lower quality bar for experimental or niche components.
+5. Where there's a component that meet the above criteria, but that isn't something we expect to be a high priority for the Aspire committers to maintain, we'll ask for a plan to sustain it (eg., motivated contributors ready to fix bugs in it)
```
Comment (eerhardt) [inline src/Components/README.md:13]:
```suggestion
5. Where there's a component that meets the above criteria, but that isn't something we expect to be a high priority for the Aspire committers to maintain, we'll ask for a plan to sustain it (eg., motivated contributors ready to fix bugs in it)
```

File: src/Components/README.md [hunk 4]
```diff
@@ -2,11 +2,33 @@
...
+
+Note: only components that are built from dotnet/aspire will be able to use the Aspire package name prefix. There is no technical barrier to using components built elsewhere, without the Aspire prefix, but currently our idea is that all broadly available Aspire components will live here in dotnet/aspire and have the Aspire package prefix. We welcome feedback on this and all the other principles listed here, though.
```
Comment (eerhardt) [inline src/Components/README.md:15]: This is a "point in time" limitation. For example, I expect the Azure components to be built from the Azure repo (which will make everyone's lives easier).

**Review by eerhardt — APPROVED** [review]

File: src/Components/README.md (hunk 4 above)
Comment (danmoseley) [reply src/Components/README.md:15]: With the Aspire prefix? So we'll still retain control and expect the prefix, but allow it on packages from certain repos. Basically have trusted packages not trusted package repos.

Comment (eerhardt) [reply src/Components/README.md:15]: > With the Aspire prefix?

Yes. I don't expect us to rename the Azure components when they move been repos.

Comment (DamianEdwards) [reply src/Components/README.md:15]: We only have thoughts beyond the current model, so I'd rather not add any text that looks like we've already committed to how a more distributed model for "official" Aspire components could look in the future.

Comment (danmoseley) [reply src/Components/README.md:15]: OK I'll merge this as is and no doubt it will evolve

**Pull request merged**